applications/GeoMechanicsApplication/tests/cpp_tests/custom_utilities/test_neighbouring_element_finder.cpp

```diff
@@ -0,0 +1,268 @@
+                          KratosGeoMechanicsFastSuiteWithoutKernel)
+{
+    Model model;
+
+    auto& r_model_part_for_entities_for_finding =
+        ModelSetupUtilities::CreateModelPartWithASingle2D3NElement(model);
+
+    auto& r_model_part_for_neighbouring_elements = model.CreateModelPart("Main");
+
+    std::vector<std::size_t> neighbour_node_ids = {1, 2};
+    PointerVector<Node>      neighbour_nodes =
+        GetNodesFromIds(r_model_part_for_entities_for_finding, neighbour_node_ids);
+    auto p_element = ElementSetupUtilities::Create2D2NElement(neighbour_nodes, {});
+    p_element->SetId(42);
+    r_model_part_for_neighbouring_elements.AddElement(p_element);
+
+    std::map<std::size_t, std::unique_ptr<BoundaryGenerator>> boundary_generators;
+    boundary_generators[std::size_t{2}] = std::make_unique<EdgesGenerator>();
+    NeighbouringElementFinder finder;
+    finder.FindEntityNeighbours(r_model_part_for_entities_for_finding.Elements(),
+                                r_model_part_for_neighbouring_elements.Elements(), boundary_generators);
+
+    // The second run should clear the previous results and find the same neighbour again.
+    // This would happen when running a multi-stage analysis (where the search is called
+    // at the start of each stage).
+    finder.FindEntityNeighbours(r_model_part_for_entities_for_finding.Elements(),
+                                r_model_part_for_neighbouring_elements.Elements(), boundary_generators);
+
+    ASSERT_EQ(r_model_part_for_entities_for_finding.GetElement(1).GetValue(NEIGHBOUR_ELEMENTS).size(), 1);
+    EXPECT_EQ(r_model_part_for_entities_for_finding.GetElement(1).GetValue(NEIGHBOUR_ELEMENTS)[0].GetId(), 42);
+}
+
+KRATOS_TEST_CASE_IN_SUITE(NeighbouringElementFinder_NeverRefersToItselfAsNeighbour, KratosGeoMechanicsFastSuiteWithoutKernel)
+{
+    Model model;
+
+    auto& r_model_part_for_entities_for_finding =
+        ModelSetupUtilities::CreateModelPartWithASingle2D3NElement(model);
+
+    std::map<std::size_t, std::unique_ptr<BoundaryGenerator>> boundary_generators;
+    boundary_generators[std::size_t{2}] = std::make_unique<EdgesGenerator>();
+    NeighbouringElementFinder finder;
+    finder.FindEntityNeighbours(r_model_part_for_entities_for_finding.Elements(),
+                                r_model_part_for_entities_for_finding.Elements(), boundary_generators);
+
+    ASSERT_EQ(r_model_part_for_entities_for_finding.GetElement(1).GetValue(NEIGHBOUR_ELEMENTS).size(), 0);
+}
+
+KRATOS_TEST_CASE_IN_SUITE(NeighbouringElementFinder_FindsNeighboursBetweenTwoContinua, KratosGeoMechanicsFastSuiteWithoutKernel)
+{
+    Model model;
+    auto& r_model_part = model.CreateModelPart("Main");
+    CreateNumberOfNewNodes(r_model_part, 4);
+
+    std::vector<std::size_t> node_ids_element_1 = {1, 2, 3};
+    PointerVector<Node>      nodes_element_1    = GetNodesFromIds(r_model_part, node_ids_element_1);
```

Comment:
```suggestion
    const auto nodes_element_1    = GetNodesFromIds(r_model_part, node_ids_element_1);
```
This suggestion also holds for `nodes_element_2`.

Reply:
Changed the PointerVector<Node> instances to const auto